browser/ai_chat/upload_file_helper_unittest.cc

```diff
@@ -0,0 +1,182 @@
+#if BUILDFLAG(IS_ANDROID)
+    testing_local_state_->Get()->registry()->RegisterBooleanPref(
+        prefs::kAllowFileSelectionDialogs, true);
+#endif
+    // To fulfill ChromeSelectFilePolicy::CanOpenSelectFileDialog()
+    testing_local_state_->Get()->SetBoolean(prefs::kAllowFileSelectionDialogs,
+                                            true);
+    file_helper_ = std::make_unique<UploadFileHelper>(web_contents(), profile);
+  }
+
+  mojom::UploadedImagePtr UploadImageSync() {
+    base::test::TestFuture<mojom::UploadedImagePtr> future;
+    file_helper_->UploadImage(
+        std::make_unique<ChromeSelectFilePolicy>(web_contents()),
+        future.GetCallback());
+    return future.Take();
+  }
+
+  void TearDown() override {
+    file_helper_.reset();
+    content::RenderViewHostTestHarness::TearDown();
+    ui::SelectFileDialog::SetFactory(nullptr);
+  }
+
+  std::unique_ptr<content::BrowserContext> CreateBrowserContext() override {
+    return std::make_unique<TestingProfile>();
+  }
+
+  std::optional<std::vector<mojom::UploadedImagePtr>> GetUploadedImages() {
+    return file_helper_->GetUploadedImages();
+  }
+
+  size_t GetUploadedImagesSize() {
+    return file_helper_->GetUploadedImagesSize();
+  }
+
+  void ClearUploadedImages() { file_helper_->ClearUploadedImages(); }
+
+ protected:
+  base::ScopedTempDir temp_dir_;
+  data_decoder::test::InProcessDataDecoder data_decoder_;
+  std::unique_ptr<ScopedTestingLocalState> testing_local_state_;
+  std::unique_ptr<UploadFileHelper> file_helper_;
+  // Must persist throughout TearDown().
+  content::SelectFileDialogParams dialog_params_;
+};
+
+TEST_F(UploadFileHelperTest, AcceptedFileExtensions) {
+  ui::SelectFileDialog::SetFactory(
+      std::make_unique<content::CancellingSelectFileDialogFactory>(
+          &dialog_params_));
+  // This also test cancel selection result
+  EXPECT_FALSE(UploadImageSync());
+  EXPECT_EQ(dialog_params_.type, ui::SelectFileDialog::SELECT_OPEN_FILE);
+  ASSERT_TRUE(dialog_params_.file_types);
+  ASSERT_EQ(1u, dialog_params_.file_types->extensions.size());
+  EXPECT_TRUE(base::Contains(dialog_params_.file_types->extensions[0],
+                             FILE_PATH_LITERAL("png")));
+  EXPECT_TRUE(base::Contains(dialog_params_.file_types->extensions[0],
+                             FILE_PATH_LITERAL("jpeg")));
+  EXPECT_TRUE(base::Contains(dialog_params_.file_types->extensions[0],
+                             FILE_PATH_LITERAL("jpg")));
+  EXPECT_TRUE(base::Contains(dialog_params_.file_types->extensions[0],
+                             FILE_PATH_LITERAL("webp")));
+#if BUILDFLAG(IS_ANDROID)
+  EXPECT_THAT(dialog_params_.accept_types,
+              testing::UnorderedElementsAre(u"image/png", u"image/jpeg",
+                                            u"image/jpg", u"image/webp"));
+#endif
+}
+
+TEST_F(UploadFileHelperTest, ImageRead) {
+  base::FilePath path = temp_dir_.GetPath().AppendASCII("not_png.png");
```

Comment:
I think missing tests for 0 byte file too

Reply:
yes, otherwise https://github.com/brave/brave-core/pull/27787/files/2a54f2f51659a81c1f2d545ec6a3b5dfefe8f003..695f947012f1e41f32d75114892a27196361917a#r1979956950 should have been caught.

Reply:
It should have the same path of early return as non-image file but it is better to add that too

Reply:
added in f7bd2eb47c1b1743f6a2579c5069bcbedf029507